crates/bevy_ecs/macros/src/lib.rs

```diff
@@ -466,3 +466,35 @@ pub fn derive_system_param(input: TokenStream) -> TokenStream {
         }
     })
 }
+
+#[proc_macro_derive(SystemLabel)]
+pub fn derive_system_label(input: TokenStream) -> TokenStream {
+    let input = parse_macro_input!(input as DeriveInput);
+    derive_label(input, Ident::new("SystemLabelMarker", Span::call_site())).into()
+}
+
+#[proc_macro_derive(StageLabel)]
+pub fn derive_stage_label(input: TokenStream) -> TokenStream {
+    let input = parse_macro_input!(input as DeriveInput);
+    derive_label(input, Ident::new("StageLabelMarker", Span::call_site())).into()
+}
+
+fn derive_label(input: DeriveInput, label_type: Ident) -> TokenStream2 {
+    let ident = input.ident;
+
+    let manifest = Manifest::new().unwrap();
```

Comment:
This won't work for examples and doc tests if `bevy_ecs` is used in the main dependencies and `bevy` in the `dev-dependencies`.
Searching in the main deps first and otherwise in the dev-dependencies like [here](https://github.com/bevyengine/bevy/pull/1426/files#diff-5ef94ef273f2d90770db3f44049b344e067a2e36349fba82112dd389b2ad803fR453) should work better.

Reply:
[This test](https://github.com/TheRawMeatball/bevy/blob/zst-labels/crates/bevy_ecs/src/schedule/system_descriptor.rs#L19-L31) passes.

Reply:
If `bevy` was added to the dev-dependencies it would fail. Or rather, the test would work but the other derives would fail.

Reply:
yeah its probably worth adopting that method here.